Add the ability for Agent to host and read from HTTPS built in integrations (#418)
grafana/agent: docs/configuration-reference.md

[instance_mode: <string> | default = "shared"]
```

### server_tls_config

> The `http_tls_config` block configures server metrics via TLS, with or without certificate pinning. In 

Reviewer: ```suggestion
The `http_tls_config` block configures the server to run with TLS.
```

> conjunction with the `integrations client_tls_config`.  Acceptable values for 

Reviewer: Use the path to the field here, not the type: 

```suggestion
conjunction with the `integrations.http_tls_config`.  Acceptable values for 
```

> conjunction with the `integrations client_tls_config`.  Acceptable values for 

Reviewer: ```suggestion
When set, `integrations.http_tls_config` must also be provided. Acceptable values for
```

> `client_auth_type` are found in [tls.config](https://golang.org/pkg/crypto/tls/#ClientAuthType). 

Reviewer: ```suggestion
`client_auth_type` are found in [Go's `tls` package](https://golang.org/pkg/crypto/tls/#ClientAuthType). 
```

[instance_mode: <string> | default = "shared"]
```

### http_tls_config

The `http_tls_config` block configures server and reading metrics via TLS, with or without certificate pinning. In 
conjunction with the `integrations client_cert_file` and `integrations client_key_file`.  Acceptable values for
`client_auth_type` are found in [tls.config](https://golang.org/pkg/crypto/tls/#ClientAuthType). 

```yaml
# File path to the server certificate
[cert_file: <string>]

# File path to the server key file
[key_file: <string>]

# Tells the server what is acceptable from the client. 

> # *NOTE* NoClientCert is not acceptable. If you want the server to accept any connection use `RequestClientCert`

Reviewer: from offline discussion: this line can be removed once we stop syncing server client/client client cert settings.

Author: Removed/Intent change, separated out the TLS configs

  metric_relabel_configs:
    [ - <relabel_config> ... ]

# Client TLS Configuration

> http_tls_config: <client_tls_config>

Reviewer: ```suggestion
http_tls_config: <tls_config>
```

  - [<remote_write>]
```

### client_tls_config

The `client_tls_config` block configures the client for tls in conjunction with the `server_tls_config`.
Values need to be defined if the server is requesting a certificate 
(Client Auth Type = RequireAndVerifyClientCert || RequireAnyClientCert). 

```yaml
# File path for the client cert
[cert_file: <string>]
  
# File path for the client key
[key_file: <string>]
  
# File path to client CA certificate
[ca_file: <string>]
```

> 

Reviewer: This is already documented by `tls_config`. I also think we can move the warning about requiring the settings to the `integrations.http_tls_config` field's comment.

Author: Good idea and done
